File: DsWorkNet/Dswork.Spring/BeanFactory.cs
using System;

using Spring.Context.Support;

namespace Dswork.Spring
{
	/// <summary>
	/// 从ApplicationContext取得spring管理的类
	/// </summary>
	public class BeanFactory
	{
		/// <summary>
		/// 取得Spring托管的类
		/// </summary>
		/// <param name="name">托管类的名称</param>
		/// <returns>Object</returns>
		public static Object GetObject(String name)
		{
			return ContextRegistry.GetContext().GetObject(name);
		}

		/// <summary>
		/// 取得Spring托管的类
		/// </summary>
		/// <typeparam name="T">Object</typeparam>
		/// <param name="name">托管类的名称</param>
		/// <returns>T</returns>
		public static T GetObject<T>(String name)
		{
			return ContextRegistry.GetContext().GetObject<T>(name);
		}
    }
}

File: DsWorkAndroid/WebApps/paper/src/gwen/devwork/BeanFactory.java
package gwen.devwork;

/**
 * 从ApplicationContext取得spring管理的类
 */
@SuppressWarnings("all")
public class BeanFactory
{
	/**
	 * 取得Spring托管的类
	 * @param name 托管类的Class
	 * @return Object
	 */
	public static Object getBean(Class name)
	{
		return org.springframework.web.context.ContextLoaderListener.getCurrentWebApplicationContext().getBean(name);
	}
	
	/**
	 * 取得Spring托管的类
	 * @param name 托管类的ID
	 * @return Object
	 */
	public static Object getBean(String name)
	{
		return org.springframework.web.context.ContextLoaderListener.getCurrentWebApplicationContext().getBean(name);
	}
}
